### ISSUE TITLE
knock-knock-who-there-file-compression-talk_FINAL_2.tar.trz.bz2.gz

### ISSUE THREAD

**Issue by lrlna** [lnug/speakers#107]
## abstract

Fade in. An opened terminal window. iTerm2 is the flavour. Dracula, the theme. You split the panes in two. You are at it again: a stash of audio buffers and that dream of manipulating a lossless file. With node as your language of choice you embark on the night. You manipulate, you transform, you have your byte order sorted, on their own accord your fingers start to type `gzip -f yourFavPhilCollinsSong.wav`. All of a sudden, a whisper: 

*But what if*.

we **stream**.
 
*file compression*.

on the **fly**.

Fade to black. 

In this talk we will walk through file compression algorithms in node as well compression standards. We will cover working with streams, audio buffers, and typed arrays to get us to compress and decompress files, and yourFavPhilCollinsSong.wav.

## bio 

tl;dr javascript, infrastructure, 🍜 🍜 🍜.

The not so short: [Irina](twitter.com/_lrlna) is a Berlin via London via Vancouver (geeeez grrrl) software developer who is, oh hey, hello, really into node.js. She writes node over at scripto.cc and [contributes to a few open-source projects](https://github.com/lrlna) in the meantime. Outside of coding, she spends quite a bit of her time exploring the outdoors, gushing over trains, and reading some Beatniks.

**Comment by claireinez** [comment]
Awesome, really looking forward to this @lrlna! @admataz can we schedule Irina in for the 28th?

**Comment by admataz** [comment]
sheduled and confirmed! 

Hi @lrlna  - looking forward to hearing your talk.

**Comment by admataz** [comment]
Hi @lrlna   - hope all is good to go for tomorrow? I assume you'll have any slides on a laptop? Usually we manage to work things out with all the projector adaptors and things for filming and recording the slides - but if you have one for your laptop to HDMI, please bring it in case...

I should be at Makers Academy from about 5:45 to get things set up - but if you arrive earlier, someone from Makers will let you in  - just tell them you're there for LNUG.  

In case you haven't got the address detail, it's 

MAKERS ACADEMY
50 - 52 COMMERCIAL ST
LONDON E1 6LT

My number is 07757234443 if you need to call for anything.

**Comment by lrlna** [comment]
hey! yep, all good for tomorrow! What time do you start, or rather what
time should I be there?

I just have a mac with thunderbolt and hdmi, so i think most places should
be ok with that!

see you tomorrow ^____^

**Comment by admataz** [comment]
If you can get there for 6:30pm that would be great - talks start at 7pm - do you have a preference for speaking earlier or later?

**Comment by lrlna** [comment]
Awesome!

And I don't mind going earlier :)

**Comment by admataz** [comment]
Hi @lrlna  - thanks again for a great talk last night and for coming to LNUG to present it. It was a great technical deep-dive, and the feedback I've had from members has been 100% good. 

Drop by next  time you're in London - hopefully it will coincide with a meetup date.  And if you ever have another talk you'd like to give I'm sure we'd gladly host it again. 

I'll keep this issue open until the video is ready and uploaded to our YouTube channel and I'll post the link here

**Comment by lrlna** [comment]
Heyyy @admataz thanks for having me! I love giving this talk, so I am glad folks enjoyed it. If you didn't see on twitter, I released [my slides](https://lrlna.github.io/talks/lnug-june-2017/#0); I am not sure if the videographer is able to get them properly with the video, but hopefully it works out :sparkles:

**Comment by admataz** [comment]
Hey @lrlna   - at last got the video last week and uploaded here: https://www.youtube.com/watch?v=Gbq0NRGo-zM

I think it worked out well and the videographer did a great job considering the technical challenges he faced. 

Thanks again for your great talk!

**Comment by lrlna** [comment]
Thanks Adam! ✨
